pkg/managed/deployment.go

```diff
@@ -73,8 +76,105 @@ func CreateDeployments(mbPair *types.ModelBindingPair, filteredConnections map[s
 	return nil
 }
 
-// Constructs the necessary Deployments for the specified ManagedCluster in the given VerrazzanoBinding
-func newDeployments(binding *v1beta1v8o.VerrazzanoBinding, managedCluster *types.ManagedCluster, manifest *util.Manifest, verrazzanoURI string, sec monitoring.Secrets) ([]*appsv1.Deployment, error) {
+// DeleteDeployments deletes deployments for a given binding.
+func DeleteDeployments(mbPair *types.ModelBindingPair, filteredConnections map[string]*util.ManagedClusterConnection) error {
+	glog.V(6).Infof("Deleting Deployments for VerrazzanoBinding %s", mbPair.Binding.Name)
+
+	// Delete Deployments associated with the given VerrazzanoBinding (based on labels selectors)
+	for clusterName, managedClusterConnection := range filteredConnections {
+		managedClusterConnection.Lock.RLock()
+		defer managedClusterConnection.Lock.RUnlock()
```

Comment:
Not sure why we need to lock, but maybe the lock is held too long.  It is not released until DeleteDeployments is returned, is that right?  Anyway, this seems to be a common pattern in the operator that should be revisited.